OpenRA.Test/OpenRA.Game/MiniYamlTest.cs

```diff
@@ -367,8 +435,69 @@ public void ChildCanBeOverriddenThenRemoved()
 				"MockString value should have been removed, but was not.");
 		}
 
-		[TestCase(TestName = "Child subnode can be removed and immediately overridden")]
-		public void ChildSubNodeCanBeRemovedAndImmediatelyOverridden()
+		[TestCase(TestName = "Merged child subnode can be removed and immediately overridden")]
+		public void MergedChildSubNodeCanBeRemovedAndImmediatelyOverridden()
+		{
+			const string BaseYaml = @"
+Test:
+	MockString:
+		CollectionOfStrings:
+			StringA: A
+			StringB: B
+Test:
+    MockString:
+		-CollectionOfStrings:
+		CollectionOfStrings:
+			StringC: C
+";
+
+			var merged = MiniYaml.Merge(new[] { BaseYaml }.Select(s => MiniYaml.FromString(s, "")))
+				.First(n => n.Key == "Test");
+
+			var traitNode = merged.Value.Nodes.Single();
+			var fieldNodes = traitNode.Value.Nodes;
+			var fieldSubNodes = fieldNodes.Single().Value.Nodes;
+
+			Assert.That(fieldSubNodes.Length == 1, Is.True, "Collection of strings should only contain the overriding subnode.");
+			Assert.That(fieldSubNodes.Single(n => n.Key == "StringC").Value.Value == "C", Is.True,
+				"CollectionOfStrings value has not been set with the correct override value for StringC.");
+		}
+
+		[TestCase(TestName = "Merged child subnode can be removed and later overridden")]
+		public void MergedChildSubNodeCanBeRemovedAndLaterOverridden()
+		{
+			const string BaseYaml = @"
+Test:
+	MockString:
+		CollectionOfStrings:
+			StringA: A
+			StringB: B
+Test:
+    MockString:
```

Comment:
this is using spaces. It's true all over this file

Reply:
Changed spaces -> tabs, aside from the test cases that specifically deal in tabs/spaces/whitespace.